### PR TITLE
Fix: Completed lic7 through usage of magnitude as distance Issue #18

### PR DESCRIPTION
**Pull request by Darpos**

### CODE REVIEW

File: src/main/CMV.java [hunk 1]
```diff
@@ -119,8 +119,24 @@ private Boolean lic6_calculate() {
...
+        if(datapoints.length < 3){
+            return 0;
+        }
+        for(int i = 0; i < datapoints.length; i++){
+            if(i + K_PTS > datapoints.length - 1){
+                return 0;
+            }
+            int j = i + K_PTS;
+            double[] vectorIJ = {datapoints[j][0] - datapoints[i][0], datapoints[j][1] - datapoints[i][1]}; 
+            double magnitudeIJ = Math.sqrt(Math.pow(vectorIJ[0],2) + Math.pow(vectorIJ[1],2));
```
Comment (a-kbd) [inline src/main/CMV.java:134]: there is a euclidean_distance function you could've used

File: src/main/CMV.java [hunk 2]
```diff
@@ -119,8 +119,24 @@ private Boolean lic6_calculate() {
...
+        final int K_PTS = Parameters.K_PTS;
+        final double LENGTH1 = Parameters.LENGTH1;
+        if(datapoints.length < 3){
+            return 0;
+        }
+        for(int i = 0; i < datapoints.length; i++){
+            if(i + K_PTS > datapoints.length - 1){
+                return 0;
+            }
+            int j = i + K_PTS;
```
Comment (a-kbd) [inline src/main/CMV.java:132]: I believe it should be j=i+K_PTS+1 cause they should be separated by that many intervening points. If i=0, j will be at index K_PTS and there would only be K_PTS - 1 points in between

File: src/main/CMV.java (hunk 1 above)
Comment (a-kbd) [inline src/main/CMV.java:134]: there is a euclidean_distance function you could've used, not super important though

**Review by a-kbd — CHANGES_REQUESTED** [review]
see comments, also I believe I read Anders said LICs shouls stil lreturn boolean

File: src/junit/LIC_test.java [hunk 3]
```diff
@@ -212,6 +212,30 @@ public void test_lic6_valid_points() {
         assertEquals(true, cmv.get_cmv_value(6));
     }
 
+    
+    /* LIC 7
+    *  K_PTS = 3
+    *  LENGTH1 = 2
+    */
+    @Test
+    public void checkIfValidGivesTrueLic7(){
```
Comment (a-kbd) [inline src/junit/LIC_test.java:221]: This test fails for me

**Review by a-kbd — CHANGES_REQUESTED** [review]

**Comment by Darpos** [comment]
Chill with this one, has some bugs

**Comment by Darpos** [comment]
Ready for acceptance.

**Review by a-kbd — APPROVED** [review]
You could've used the euclidean_distance helper function directly instead of vector_subtraction→vector_magnitude but it still works and all tests pass so looks good to me :+1:

**Pull request merged**